pandas/_libs/lib.pyx

```diff
@@ -375,6 +375,23 @@ cpdef ndarray[object] list_to_object_array(list obj):
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
+cpdef ndarray[object] tuple_to_object_array(tuple obj):
```

Comment:
don't do this, you are dupliating lots of code. I suspect you are hitting

```
In [2]: pd.unique(('foo'),)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-bd2fadf86f43> in <module>()
----> 1 pd.unique(('foo'),)

/Users/jreback/pandas/pandas/core/algorithms.py in unique(values)
    348     """
    349 
--> 350     values = _ensure_arraylike(values)
    351 
    352     # categorical is a fast-path

/Users/jreback/pandas/pandas/core/algorithms.py in _ensure_arraylike(values)
    171         inferred = lib.infer_dtype(values)
    172         if inferred in ['mixed', 'string', 'unicode']:
--> 173             values = lib.list_to_object_array(values)
    174         else:
    175             values = np.asarray(values)

TypeError: Argument 'obj' has incorrect type (expected list, got str)

```

if so, pls do a separate PR for this (it won't involve *any* cython code, just a simple change).